kani-compiler/src/kani_middle/mod.rs

```diff
@@ -50,6 +54,23 @@ pub fn check_crate_items(tcx: TyCtxt, ignore_asm: bool) {
     tcx.sess.abort_if_errors();
 }
 
+/// Check that all given items are supported and there's no misconfiguration.
+/// This method will exhaustively print any error / warning and it will abort at the end if any
+/// error was found.
+pub fn check_reachable_items(tcx: TyCtxt, queries: &QueryDb, items: &[MonoItem]) {
+    // Avoid printing the same error multiple times for different instantiations of the same item.
+    let mut def_ids = HashSet::new();
+    for item in items {
+        let def_id = item.def_id();
+        if !def_ids.contains(&def_id) {
+            // Check if any unstable attribute was reached.
+            check_unstable_features(tcx, queries.get_unstable_features(), def_id);
```

Comment:
It's assumed here that two or more instantiations of the same item don't need to be checked. Why are they given then? How can they differ?

Reply:
Because generic items have the same `def_id` and they will have the same attributes attached to them. The input to this function is working on the top of monomorphized items though, so we really only need to check the attributes once.